pkg/scanners/clairv4/clairv4.go

```diff
@@ -199,37 +205,38 @@ func (c *clairv4) index(manifest *claircore.Manifest) error {
 		return err
 	}
 
-	// TODO: add retries
-
-	req, err := http.NewRequest(http.MethodPost, c.indexEndpoint, bytes.NewReader(body))
-	if err != nil {
-		return err
-	}
-
-	resp, err := c.client.Do(req)
-	if err != nil {
-		return err
-	}
-	defer utils.IgnoreError(resp.Body.Close)
+	return retry.WithRetry(func() error {
+		// Make a new request per retry to ensure the body is fully populated for each retry.
```

Comment:
Dug into this a bit, nice catch - the need to re-create a request seems as if it would be a common cause for bugs. Interestingly I see that client.Do will ['reset'](https://cs.opensource.google/go/go/+/refs/tags/go1.19.5:src/net/http/client.go;l=662-669;drc=1c65b69bd1dbc930c6246877f6c21c81f2a60d55;bpv=1;bpt=1) the body reader for certain types, but it only does that when following redirects, not errors.

Reply:
I only caught this because I ran into this error once when testing and looked it up 😂